include/aikido/robot/Robot.hpp

```diff
@@ -2,93 +2,345 @@
+
+  ///
+  /// Given a collision constraint, returns it combined with the self-collision
+  /// constraint of the entire robot.
+  /// \param[in] collisionFree The collision constraint that is to be tied with
+  /// root robot's self-collision constraint.
+  constraint::TestablePtr combineCollisionConstraint(
+      const constraint::dart::CollisionFreePtr& collisionFree) const;
+
+  ///
+  /// Get the collission constraint between the root robot
+  /// and selected bodies within its World, combined with
+  /// its self-collision constraint.
+  /// \param[in] bodyNames Names of the bodies in
+  /// the robot's world to check collision with.
+  /// Leave blank to check with all non-robot bodies.
+  constraint::TestablePtr getWorldCollisionConstraint(
+      const std::vector<std::string> bodyNames
+      = std::vector<std::string>()) const;
+
+  ///
+  /// Registers a subset of the joints of the skeleton as a new robot.
+  /// Must be disjoint from other subrobots.
+  ///
+  /// \param[in] metaSkeleton The metaskeleton corresponding to the subrobot.
+  /// \param[in] name Name of the subrobot.
+  virtual RobotPtr registerSubRobot(
       const dart::dynamics::MetaSkeletonPtr& metaSkeleton,
-      const constraint::dart::CollisionFreePtr& collisionFree) const = 0;
+      const std::string& name);
+
+  ///
+  /// Get existing subrobot pointer. Or nullptr if not found.
+  ///
+  /// \param[in] name Name of the subrobot.
+  RobotPtr getSubRobotByName(const std::string& name) const;
+
+  ///
+  /// Plan the robot to a specific configuration.
+  ///
+  /// \param[in] goalState Goal configuration.
+  /// \param[in] testableConstraint Planning (e.g. collision) constraints, set
+  /// to nullptr for no constraints (not recommended)
+  /// \param[in] planner Configuration planner, defaults to Snap Planner
+  trajectory::TrajectoryPtr planToConfiguration(
+      const statespace::StateSpace::State* goalState,
+      const constraint::TestablePtr& testableConstraint,
+      const std::shared_ptr<planner::ConfigurationToConfigurationPlanner>&
+          planner
+      = nullptr) const;
+
+  /// Default to selfCollisionConstraint
+  trajectory::TrajectoryPtr planToConfiguration(
+      const statespace::StateSpace::State* goalState,
+      const std::shared_ptr<planner::ConfigurationToConfigurationPlanner>&
+          planner
+      = nullptr) const
+  {
+    return planToConfiguration(
+        goalState, getSelfCollisionConstraint(), planner);
+  }
+
+  ///
+  /// Plan a specific body node of the robot to a
+  /// sample configuraiton within a Task Space Region.
+  ///
+  /// \param[in] bodyNodeName Bodynode (usually the end effector) whose frame
+  /// should end up in the TSR.
+  /// \param[in] tsr \see constraint::dart::TSR
+  /// \param[in] testableConstraint Planning (e.g. collision) constraints,
+  /// set to nullptr for no constraints (not recommended)
+  /// \param[in] maxSamples Maximum number of TSR samples
```

Comment:
Just flagging that this will be a merge conflict with https://github.com/personalrobotics/aikido/pull/607 . No worries -- whoever merges second can adapt to whoever merges first, but I just wanted to flag this for our records.